pkg/deploy/deploy.go

```diff
@@ -84,27 +84,35 @@ func Deploy(ctx context.Context, projects []project.Project, environmentClients
 		}
 		errors.As(validationErrs, &deploymentErrs)
 	}
+
+	reporter := report.GetReporterFromContextOrDiscard(ctx)
+
+	preloadCaches(ctx, projects, environmentClients)
```

Comment:
From a performance point of view, does it make sense to preload the caches (which can take a long time) after building the graph and sorting the configs?

Reply:
Yes, you are right. It would be better to first validate the cyclic dependency and only after that do the preload of the cache. Adjusted 👍